TypeScript.YAML-tmLanguage

```diff
@@ -154,15 +154,15 @@ repository:
 
   method-declaration:
     name: meta.method.declaration.ts
-    begin: '\b(?:(abstract)\s+)?\b(?:(public|private|protected)\s+)?\b(?:(async)\s+)?(?:(get|set)\s+)?(?:(new)|(?:\b(constructor)\b)|(?:([a-zA-Z_$][\.\w$]*)\s*(\??)))?\s*(?=\(|\<)'
+    begin: '\b(?:(abstract)\s+)?\b(?:(public|private|protected)\s+)?\b(?:(async)\s+)?(?:(get|set)\s+)?(?:(new)|(?:\b(constructor)\b)|(?:(\[*)(`|'|"*)(.*)(`|'|"*)(\]*)\s*(\??)))?\s*(?=\(|\<)'
```

Comment:
This still doesn't look correct. For example, the `.*` would eat everything including the closing quote and bracket (the test case needs to be updated for it). Also, do you need to consider if the two quotes are matching? For example, `["test']` and `['test"]` shouldn't be treated the same way as the `["test"]`